### PR TITLE
[scoped-custom-element-registry] Improve issues related to `formAssociated`

### LINKED ISSUE #546: [scoped-custom-element-registry]: Safari v16.4 focuses all custom elements
### Description

Safari v16.4 supports `attachInternals` and `formAssociated` features natively, and with the subject polyfill enabled _all_ custom elements on the page are keyboard-focusable using Tab key, regardless of the `role` or anything else.

Here is the root of the problem:

https://github.com/webcomponents/polyfills/blob/90cb97f847ce918289dac0978c50dcda0a0afd72/packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js#L218-L220

Feels like Safari tries to focus any element with `static formAssociated = true`. If I change that to `false` in the polyfill source code then the above wrong behavior becomes fixed.

Is there a way to respect the original `formAssociated` value instead of enforcing it to be `true`?

### Example

```html
<!DOCTYPE html>
 <html>
 <head>
   <script src="https://www.unpkg.com/@webcomponents/scoped-custom-element-registry@0.0.9/scoped-custom-element-registry.min.js"></script>
 </head>
 <body>
   <input type="text"/>
   <my-el></my-el>
 
   <script>
     class MyEl extends HTMLElement {
       constructor() {
         super()
 
         const shadow = this.attachShadow({ mode: 'open' })
 
         shadow.innerHTML = '<span>Hello</span>'
       }
     }
 
     customElements.define('my-el', MyEl)
   </script>
 </body>
 </html>
```

### Version

v0.0.9

### Browsers affected

- [ ] Chrome
- [ ] Firefox
- [ ] Edge
- [x] Safari
- [ ] IE 11

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #546, #547, #580, #570, #549. 

Note, bulk of changes in [this commit](https://github.com/webcomponents/polyfills/pull/581/commits/b0ef750fe9e4759a3c6202d05ae5c2c7e9ebe203).

The setting for a defined class' `formAssociated` cannot be properly polyfilled per registry. To address this, previously the polyfill always set the stand-in class to be `formAssociated == true`. However, this caused a series of subtle issues.

Instead, it is now set to whatever the first defined tag name's `formAssociated` is set. It's also possible to reserve a given name to always be formAssociated by adding it to `customElementRegistryPolyfill.formAssociated.add(tagName)`.

Unconditionally installs polyfill.

Adds `registry` to ShadowRootInit

### CODE REVIEW

File: packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js [hunk 1]
```diff
@@ -11,12 +11,22 @@
  * subject to an additional IP rights grant found at
  * http://polymer.github.io/PATENTS.txt
  */
-if (!ShadowRoot.prototype.createElement) {
+if (!(ShadowRoot.prototype.createElement && ShadowRoot.prototype.importNode)) {
```
Comment (thepassle) [inline packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js:14]: Should this feature check be here at all?

Comment (sorvell) [reply packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js]: Reverted for now. Will have more to change/discuss to potentially better support Chrome's partial implementation in a follow-on.

Comment (sorvell) [reply packages/scoped-custom-element-registry/src/scoped-custom-element-registry.js]: Based on dicussion with @justinfagnani and suggesting in #549, check was removed.

File: packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts [hunk 2]
```diff
@@ -19,6 +19,22 @@
...
  */
 
+interface Window {
+  CustomElementRegistryPolyfill: {
+    formAssociated: Set<string>;
+  };
+}
+
+// Polyfill helper object
+(window as Window)['CustomElementRegistryPolyfill'] = {
```
[the remark this answers is not in the document]
Comment (sorvell) [reply packages/scoped-custom-element-registry/src/scoped-custom-element-registry.ts]: Done and added comment.

**Pull request merged**